packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +61,106 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const isValidDate = (date: Date) => date instanceof Date && !isNaN(date.valueOf());
+
+const calculateTimezoneByValueOld = (timezone: string, date = new Date()) => {
+  const customTimezones = timeZoneDataUtils.getTimeZonesOld();
+  if (customTimezones.length === 0) {
+    return undefined;
+  }
+
+  const dateUtc = createUTCDate(date);
+  return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+};
+
+const calculateTimezoneByValueCore = (timeZone: string, date = new Date()): number | undefined => {
+  let offset;
+  try {
+    offset = getStringOffset(timeZone, date);
+  } catch (e) {
+    errors.log('W0009', timeZone);
+    return undefined;
+  }
+
+  if (offset === GMT) {
+    return 0;
   }
-  return timezone;
+
+  const isMinus = offset.substring(3, 4) === '-';
+  const hours = offset.substring(4, 6);
+  const minutes = offset.substring(7, 9);
+
+  const result = parseInt(hours, 10) + parseInt(minutes, 10) / MINUTES_IN_HOUR;
+  return isMinus ? -result : result;
+};
+
+const calculateTimezoneByValue = (timeZone: string | undefined, date = new Date()): number | undefined => {
```

Comment:
I see common return of undefined. May be create type Nullable?

`type Nullable<T> = T | undefined`
`const calculateTimezoneByValue = () => Nullable<number>`

@williamvinogradov @pomahtri What do uou think about it?

Reply:
I'm against, looks useless (what are advantages of `Nullable<T>` vs `T | undefined`?), plus can lead to misunderstandings (why `Nullable` contains `undefined` instead of `null`)